oap-server/server-configuration/configuration-nacos/src/test/java/org/apache/skywalking/oap/server/configuration/nacos/NacosConfigWatcherRegisterTest.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.oap.server.configuration.nacos;
+
+import com.alibaba.nacos.api.exception.NacosException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.skywalking.oap.server.configuration.api.ConfigTable;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * @author kezhenxu94
+ */
+public class NacosConfigWatcherRegisterTest {
+    @Test
+    public void shouldReadConfigs() throws NacosException {
```

Comment:
Could you learn how es client test by using docker image to do real test?

Reply:
> Could you learn how es client test by using docker image to do real test?

Sure, am I supposed to create the test in SkyAPMTest?

Reply:
No, you don't need that. See client library module. There is a way to start container in unit test.

Reply:
Hi @wu-sheng , integration tests are added, it's my very first try with the maven-docker plugin, please pay special attention to those parts, thanks very much :-)

Reply:
I run it locally and it works well and all tests passed, but travis-ci seems did not start mysql first (nacos depends on mysql):

```
[ERROR] DOCKER> [nacos/nacos-server:1.0.0] "nacos-dynamic-configuration-integration-test": Timeout after 60288 ms while waiting on log out 'Nacos started successfully'
[ERROR] DOCKER> Error occurred during container startup, shutting down...
[ERROR] DOCKER> I/O Error [[nacos/nacos-server:1.0.0] "nacos-dynamic-configuration-integration-test": Timeout after 60288 ms while waiting on log out 'Nacos started successfully']
```

Reply:
Found the cause and fixed

Reply:
Ci works now and tests passed